nuxeo-features/nuxeo-platform-oauth/src/main/java/org/nuxeo/ecm/platform/oauth2/clients/OAuth2Client.java

```diff
@@ -135,6 +183,94 @@ public boolean isValidWith(String clientId, String clientSecret) {
         return enabled && id.equals(clientId) && (StringUtils.isEmpty(secret) || secret.equals(clientSecret));
     }
 
+    /**
+     * Create {@link DocumentModel} from {@link OAuth2Client}.
+     *
+     * @param oAuth2Client the oAuth2Client to convert
+     * @return the corresponding {@code DocumentModel} of {@code OAuth2Client}
+     * @throws NullPointerException if oAuth2Client is {@code null}
+     * @throws NuxeoException if oAuth2Client is not valid
+     * @since 11.1
+     */
+    public static DocumentModel fromOAuth2Client(OAuth2Client oAuth2Client) {
+        return BaseSession.createEntryModel(null, OAUTH2CLIENT_SCHEMA, null, getDocumentData(oAuth2Client));
+    }
+
+    /**
+     * Update {@link DocumentModel} by {@link OAuth2Client}.
+     *
+     * @param documentModel the document model to update
+     * @param oAuth2Client the new values of document
+     * @return the updated {@code DocumentModel}
+     * @throws NullPointerException if the documentModel or oAuth2Client is {@code null}
+     * @throws NuxeoException if oAuth2Client is not valid
+     * @since 11.1
+     */
+    public static DocumentModel updateDocument(DocumentModel documentModel, OAuth2Client oAuth2Client) {
```

Comment:
Maybe you don't need this method and can just call ```documentModel.setProperties(OAUTH2CLIENT_SCHEMA, OAuth2Client.getDocumentData(oAuth2Client));``` in `OAuth2ClientServiceImpl#update`, WDYT?

Reply:
initially it's what i did and after i find my self doing creation by helper method(i means the one in OAuth2Client) and update calling a method with Map as returned param in service. this is way i create the update just to be consistent with what i did for creation.  

other thing is returned Map i am not really comfortable with in (another reason why i create the update). Because we dont know what's in our Map. What i did in the past when i need a Map type. I wrap it into class and i make the map private and just exposing  get, remove and update. But as we should avoid using private accessor i just create this method.